spec/models/ext_management_system_spec.rb

```diff
@@ -94,6 +94,16 @@
     expect(described_class.with_eligible_manager_types(r.class).count).to eq(1)
   end
 
+  it "validates type" do
+    v = FactoryBot.create(:ems_vmware)
```

Comment:
What do you think of tests like this `expect { ManageIQ::Providers::BaseManager.new(:hostname => "abc", :name => "abc").validate! }.to raise_error(…)` with a couple of valid and invalid EMSs?  Using factories will not allow you to test any of the non-leaf subclasses.

Reply:
Yeah, I was wondering about that but didn't want to mention a non-leaf provider class in the core repo.  Your example is much better since this class lives here.  👍 

I think that's a valid test.

Reply:
oh right ok thanks